output/json.go

```diff
@@ -24,48 +25,58 @@ type JSONReport struct {
 	Behaviors map[string]*JSONBehaviorReport `json:"behaviors"`
 }
 
-var JSON ReporterFunc = func(config *plan.Config, tests <-chan execute.TestResponse) Summary {
-	summary := Summary{}
-	report := JSONReport{
-		Behaviors: make(map[string]*JSONBehaviorReport),
+type JSON struct {
+	r    JSONReport
```

Comment:
I'm not sure the brevity is worth it here - seeing `j.r...` below is a bit confusing. Suggest `j.report...` instead.